crates/ruff_cli/src/bin/ruff.rs

```diff
@@ -0,0 +1,64 @@
+use clap::{Parser, Subcommand};
+use std::process::ExitCode;
+
+use colored::Colorize;
+
+use ruff_cli::args::{Args, Command};
+use ruff_cli::{run, ExitStatus};
+
+#[cfg(target_os = "windows")]
+#[global_allocator]
+static GLOBAL: mimalloc::MiMalloc = mimalloc::MiMalloc;
+
+#[cfg(all(
+    not(target_os = "windows"),
+    not(target_os = "openbsd"),
+    any(
+        target_arch = "x86_64",
+        target_arch = "aarch64",
+        target_arch = "powerpc64"
+    )
+))]
+#[global_allocator]
+static GLOBAL: tikv_jemallocator::Jemalloc = tikv_jemallocator::Jemalloc;
+
+pub fn main() -> ExitCode {
+    let mut args: Vec<_> = std::env::args().collect();
+
+    // Clap doesn't support default subcommands but we want to run `check` by
+    // default for convenience and backwards-compatibility, so we just
+    // preprocess the arguments accordingly before passing them to Clap.
+    if let Some(arg) = args.get(1) {
```

Comment:
Not for this PR, but i'd like it if just `ruff` ran linting from the project root pyproject.toml (or the git root), similar to `cargo fmt` or `cargo clippy`